packages/plugins/plugin-stack/src/components/AddSectionDialog.tsx

```diff
@@ -18,7 +18,9 @@ import { nonNullable } from '@dxos/util';
 import { STACK_PLUGIN } from '../meta';
 import { StackViewType, type StackPluginProvides, type AddSectionPosition } from '../types';
 
-type AddSectionDialogProps = { path?: string; position: AddSectionPosition; collection: CollectionType };
+export const ADD_SECTION_DIALOG = `${STACK_PLUGIN}/AddSectionDialog`;
+
+export type AddSectionDialogProps = { path?: string; position: AddSectionPosition; collection: CollectionType };
 
 export const dataHasAddSectionDialogProps = (data: any): data is { subject: AddSectionDialogProps } => {
   return (
```

Comment:
i think we can ultimately move away from using these "sniffers" by using effect schema for all types.

Reply:
> for all types

maybe, though as @dmaretskyi has pointed out, effect schema's don't boil away like TS types do so they do make the app bigger. I think there's lots of types where the schema wouldn't be that useful

> ultimately move away from using these "sniffers"

I do think that we could have typed surfaces so that the data passed in is type-checked too, similar to what I'm doing w/ intents. This cuts down on a lot of the need for guards/data parsing, however I'm still finding some cases where I wish I had the guards in the intents (e.g. schema's w/ optional fields) so I suspect they'll still around and probably be ported over to intents as well, but they'll be relied on less heavily.